### PR TITLE
added the publish wrapper

### PR DESCRIPTION
**Pull request by isaac-darling**
DO-NOT-MERGE is added while `dl.easy_publish` is under development (soon to be completed).

### CODE REVIEW

**Review by ascourtas — CHANGES_REQUESTED** [review]
Hi @isaac-darling I'm not quite sure what's going on here, it looks like you've commented out the entirety of the dataset publication functionality?

**Review by ascourtas — APPROVED** [review]
Great work! Much less concerning. Conditioned on the merge of https://github.com/DLHub-Argonne/dlhub_sdk/pull/157 , `DO NOT MERGE` before that

File: foundry/foundry.py [hunk 1]
```diff
@@ -396,6 +396,28 @@ def publish(self, foundry_metadata, data_source, title, authors, update=False,
...
+            (string): task id of this submission, can be used to check for success
+        Raises:
+            ValueError: If the given servable_type is not in the list of acceptable types
+            Exception: If the serv_options are incomplete or the request to publish results in an error
+        """
+        return self.dlhub_client.easy_publish(title, creators, short_name, servable_type, serv_options)
+
     # TODO: come back and address DLHub code
     #
     # def describe_model(self):
```
Comment (ascourtas) [inline foundry/foundry.py:423]: actually, one small edit -- please remove the commented out code below, it should no longer be necessary based on https://github.com/DLHub-Argonne/dlhub_sdk/pull/157

File: foundry/foundry.py [hunk 2]
```diff
@@ -396,6 +396,28 @@ def publish(self, foundry_metadata, data_source, title, authors, update=False,
...
+            title (string): title for the servable
+            creators (string | list): either the creator's name (FamilyName, GivenName) or a list of the creators' names
+            short_name (string): shorthand name for the servable
+            servable_type (string): the type of the servable, must be a member of ("static_method",
+                                                                                   "class_method",
+                                                                                   "keras",
+                                                                                   "pytorch",
+                                                                                   "tensorflow",
+                                                                                   "sklearn")
+            serv_options (dict): the servable_type specific arguments that are necessary for publishing
```
Comment (ascourtas) [inline foundry/foundry.py:412]: please link to where the user may find the list of arguments, and how they're applied

**Review by ascourtas — COMMENTED** [review]
a couple tiny things

File: foundry/foundry.py (hunk 2 above)
Comment (isaac-darling) [reply foundry/foundry.py]: let me know if the new changes leave anything to be desired

Comment (ascourtas) [reply foundry/foundry.py]: look good!

**Review by ascourtas — APPROVED** [review]
This one is ready to go as soon as the other two dlhub_sdk PRs are good

**Comment by isaac-darling** [comment]
Tests are failing with an auth issue, I'm uncertain of the precise cause.

**Comment by ascourtas** [comment]
> Tests are failing with an auth issue, I'm uncertain of the precise cause.

I believe this is due to the funcx auth issue

**Comment by ascourtas** [comment]
once tests are resolved with funcx issue and `GenericAlias` issue is resolved in dlhub_sdk PRs, this can be merged

**Comment by ascourtas** [comment]
now just waiting on resolution of PRs in DLHub SDK (eg `GenericAlias` issue among other typing incompatibilities)

**Comment by Aadit-Ambadkar** [comment]
is this ready to be merged?

**Comment by isaac-darling** [comment]
soon, the last step on the dlhub side of things is a PR review from Logan/Ari

**Pull request merged**